lib/katello/engine.rb

```diff
@@ -99,6 +89,17 @@ class Engine < ::Rails::Engine
                         #{Katello::Engine.root}/app/lib/headpin/actions
                         #{Katello::Engine.root}/app/lib/katello/actions)
       ForemanTasks.dynflow.config.eager_load_paths.concat(action_paths)
+      ForemanTasks.dynflow.eager_load_actions!
+    end
+
+    # make sure the Katello plugin is initialized before `after_initialize`
+    # hook so that the resumed Dynflow tasks can rely on everything ready.
+    initializer 'katello.register_plugin', :before => :finisher_hook, :after => 'foreman_remote_execution.register_plugin' do
+      ::Foreman::AccessControl::Permission.prepend ::Katello::Concerns::PermissionExtensions
+      require 'katello/plugin'
+
+      # extend builtin permissions from core with new actions
+      require 'katello/permissions'
     end
 
     initializer "katello.set_dynflow_middlewares", :before => :finisher_hook do |_app|
```

Comment:
I am not sure I understand the consequences of this, but since I'd like to ask for minor release inclusion for this patch I wonder if @adamruzicka would be able to confirm this is safe to do.

Reply:
This looks reasonable to me and I don't see any immediate footguns. @jlsherrill anything I'm overlooking?